lennoxs30api/s30api_async.py

```diff
@@ -879,6 +882,28 @@ def processDevices(self, message):
             _LOGGER.error("processDevices - Exception " + str(e))
             raise S30Exception(str(e), EC_PROCESS_MESSAGE, 1)
 
+    def processEquipments(self, message) -> bool:
+        update = False
+        try:
```

Comment:
While this would work perfectly, Python has a dictionary method that enables a more compact expression of this logic:
```
def processEquipments(message):
    for equipment in message:
        for parameter in equipment.get("equipment", {}).get("parameters", []):
            if parameter.get("parameter", {}).get("name") == "Single Setpoint Mode":
                value = parameter["parameter"]["value"]
```

I'm not sure the pid is always 525 as it is in my system, and so I'd personally recommend matching by name.

Reply:
I checked the pid on my system it is also 525, the only difference is mine is marked as not visible so it does not show in the UI.   It's also in other JSON i've received from users of the app.  Since they have the S30 available in several different languages, they may be translating these names in the config?  I don't have any examples of what is looks like in a foreign language yet.

Reply:
Oh cool, that's good to know! That's probably fine then, it seems like it's meant to be a self-describing UI and so it's very possible that they'll have a translated name for other languages.

Reply:
This is clever.  Thanks for teaching me some python.  So the first get instead of returning a None if the key is not found; returns an empty set such that the second get won't error (e.g. calling get on None) and then if it doesn't find parameters it returns an empty array.  Got alot of code to simplify :-).